test/regression/2724_failing_publisher.cc

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef _WIN32
+#include <unistd.h>
+#endif
+#include "gazebo/test/ServerFixture.hh"
+
+using namespace gazebo;
+using namespace gazebo::transport;
+
+class FailingPublisherTest : public ServerFixture
+{
+};
+
+class TestTransport : public CallbackHelper
+{
+  /////////////////////////////////////////////////
+  public: virtual bool HandleData(const std::string &/*_newdata*/,
+                                  boost::function<void(uint32_t)> _cb,
+                                  uint32_t _id) override
+  {
+    _cb(_id);
+    return true;
+  }
+
+  /////////////////////////////////////////////////
+  public: virtual bool HandleMessage(MessagePtr /*_newMsg*/) override
+  {
+    return true;
+  }
+
+  /////////////////////////////////////////////////
+  public: virtual bool IsLocal() const override
+  {
+    return false;
+  }
+};
+
+int g_receivedMsgs = 0;
+
+void RequestMsgCb(const ConstRequestPtr &/*_msg*/)
+{
+  ++g_receivedMsgs;
+}
+
+/////////////////////////////////////////////////
+TEST_F(FailingPublisherTest, DirectPublish)
+{
+  Load("worlds/empty.world");
+
+  // Initialize node
+  NodePtr node = boost::make_shared<Node>();
```

Comment:
This is failing compilation on CI:

```
/var/lib/jenkins/workspace/gazebo-ci-pr_any-ubuntu_auto-amd64-gpu-nvidia/gazebo/test/regression/2724_failing_publisher.cc:67:18: error: ���make_shared��� is not a member of ���boost���
   NodePtr node = boost::make_shared<Node>();
```

I think you may need to use `new` with the supported boost version.